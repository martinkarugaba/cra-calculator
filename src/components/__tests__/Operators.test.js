import PropTypes from 'prop-types';
import { render } from '@testing-library/react';
import Operators from '../Operators';

function MockSingleButton({ buttonName, basis, bg }) {
  return (
    <div data-testid="single-button">
      `
      {buttonName}
      {' '}
      -
      {' '}
      {basis}
      {' '}
      -
      {' '}
      {bg}
      `
    </div>
  );
}

jest.mock('../SingleButton', () => MockSingleButton);

jest.mock('../../data/operators', () => [
  { id: 1, text: 'Operator 1' },
  { id: 2, text: 'Operator 2' },
  { id: 3, text: 'Operator 3' },
]);

MockSingleButton.propTypes = {
  buttonName: PropTypes.string.isRequired,
  basis: PropTypes.string.isRequired,
  bg: PropTypes.string.isRequired,
};

describe('Operators component', () => {
  it('matches the snapshot', () => {
    const { container } = render(<Operators />);
    expect(container).toMatchSnapshot();
  });

  it('renders the correct number of SingleButton components', () => {
    const { getAllByTestId } = render(<Operators />);
    const singleButtons = getAllByTestId('single-button');
    expect(singleButtons).toHaveLength(3);
  });

  it('passes the correct props to SingleButton components', () => {
    const { getAllByTestId } = render(<Operators />);
    const singleButtons = getAllByTestId('single-button');
    expect(singleButtons[0]).toHaveTextContent(
      'Operator 1 - basis-1/3 - bg-orange-400',
    );
    expect(singleButtons[1]).toHaveTextContent(
      'Operator 2 - basis-1/3 - bg-orange-400',
    );
    expect(singleButtons[2]).toHaveTextContent(
      'Operator 3 - basis-1/3 - bg-orange-400',
    );
  });
});
