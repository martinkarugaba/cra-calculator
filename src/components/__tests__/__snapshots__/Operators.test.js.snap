// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Operators component matches the snapshot 1`] = `
<div>
  <div
    class="flex flex-col justify-end w-[25%] h-[300px]"
  >
    <div
      data-testid="single-button"
    >
      Operator 1
       - 
      basis-1/3
       - 
      bg-orange-400
    </div>
    <div
      data-testid="single-button"
    >
      Operator 2
       - 
      basis-1/3
       - 
      bg-orange-400
    </div>
    <div
      data-testid="single-button"
    >
      Operator 3
       - 
      basis-1/3
       - 
      bg-orange-400
    </div>
  </div>
</div>
`;
